decidim-core/app/models/decidim/scope.rb

```diff
@@ -10,6 +10,70 @@ class Scope < ApplicationRecord
                class_name: "Decidim::Organization",
                inverse_of: :scopes
 
-    validates :name, presence: true, uniqueness: { scope: :organization }
+    belongs_to  :scope_type,
+                foreign_key: "scope_type_id",
+                class_name: "Decidim::ScopeType",
+                inverse_of: :scopes
+
+    belongs_to  :parent,
+                foreign_key: "parent_id",
+                class_name: "Decidim::Scope",
+                inverse_of: :children,
+                optional: true
+
+    has_many    :children,
+                foreign_key: "parent_id",
+                class_name: "Decidim::Scope",
+                inverse_of: :parent
+
+    validates :name, :code, :organization, presence: true
+    validates :code, uniqueness: { scope: :organization }
+    validate :forbid_cycles
+    after_create_commit :create_part_of
+    before_validation :update_part_of, on: :update
+
+    # Scope to return only the top level scopes.
+    #
+    # Returns an ActiveRecord::Relation.
+    def self.top_level
+      where(parent_id: nil)
+    end
+
+    def siblings
+      if parent
+        parent.children
+      else
+        organization.scopes.top_level
+      end
+    end
+
+    def part_of_scopes
+      Decidim::Scope.where(id: part_of)
```

Comment:
It is not a scope, it is an instance method that retrieves all part_of scope objects.